src/rez/solver.py

```diff
@@ -1946,14 +2004,16 @@ def __init__(self, package_requests, package_paths, context=None,
         else:
             self.optimised = optimised
 
-        self.phase_stack = None
-        self.failed_phase_list = None
-        self.abort_reason = None
-        self.callback_return = None
-        self.depth_counts = None
-        self.solve_begun = None
-        self.solve_time = None
-        self.load_time = None
+        # these values are all set in _init()
+        self.phase_stack: list[_ResolvePhase]
+        self.failed_phase_list: list[_ResolvePhase]
+        self.depth_counts: dict
+        self.solve_begun: bool
+        self.solve_time: float
+        self.load_time: float
```

Comment:
these values are never actually `None`, so we define the types here, and their values are assigned in `_init()`